remote oauth (#9127)
fishtown-analytics/docs.getdbt.com: website/docs/docs/platform/manage-access/connect-apps-oauth.md

Dynamically registered apps appear in the **Dynamically registered** table, which shows:

| Column | Description |
|--------|-------------|
| Name | The app name |
| Redirect URL | The OAuth callback URL the app registered |
| Created on | When the client was first registered |
| Last used on | When the client last made an authenticated request |

<!-- add info on how to disable dynamic client registration when it's live -->

### Manual registration

For apps that don't support dynamic registration (for example, Salesforce agent or custom-built integrations), an admin can manually register an OAuth client.

To add a manually registered integration:

1. Go to **Account settings** → **Integrations** → **App integrations**.
2. Click **+ Add integration**.
3. Complete the registration form. Each manually registered client receives a unique client ID. Note that no client secret is issued, and the client must instead support the [PKCE extension (RFC 7636)](https://datatracker.ietf.org/doc/html/rfc7636).

Manually registered clients appear in the **Manually registered** section.

> When configuring your client, use the following OAuth endpoints (replace `yourprefix` with your dbt platform subdomain):

Reviewer: Should this be `YOURPREFIX` all caps to be consistent with our style guide?

> When configuring your client, use the following OAuth endpoints (replace `yourprefix` with your dbt platform subdomain):

Author: ```suggestion
When configuring your client, use the following OAuth endpoints (replace `YOUR_PREFIX` with your dbt platform subdomain):
```

| Endpoint | URL |
|----------|-----|

> | Authorization | `https://yourprefix.dbt.com/oauth/authorize` |

Reviewer: Same questions...should placeholder text be all caps?

> | Authorization | `https://yourprefix.dbt.com/oauth/authorize` |

Author: ```suggestion
| Authorization | `https://YOUR_PREFIX.dbt.com/oauth/authorize` |
```

> | Token | `https://yourprefix.dbt.com/oauth/token` |

Author: ```suggestion
| Token | `https://YOUR_PREFIX.dbt.com/oauth/token` |
```

## Use with remote MCP

When you connect an MCP client to the [remote dbt MCP server](/docs/dbt-ai/setup-remote-mcp#oauth-remote-mcp), it authenticates using OAuth. The client needs just your dbt platform MCP URL, which you can find under **Account settings** → **Access URLs** → **MCP Endpoint URL**. 

Clients that support dynamic registration complete the registration step automatically &mdash; you'll see them appear in the **Dynamically registered** table after first use.

> 1. To connect a custom MCP client to dbt, navigate to your AI tool's connector settings and enter your <Constant name="dbt_platform" /> MCP URL. The following example shows how to connect a custom MCP (Claude in this case) client to dbt. 

Author: ```suggestion
1. To connect a custom MCP client to dbt, navigate to your AI tool's connector settings and enter your <Constant name="dbt_platform" /> MCP URL. The following example shows how to connect dbt to a custom MCP in Claude Desktop.
```

  
2. Enter a name and paste your MCP URL (for example, `https://abc123.us1.dbt.com/api/ai/v1/mcp`), then click **Add**.
     <Lightbox src="/img/docs/dbt-cloud/oauth-add-custom-connector.png" title="Custom connector dialog showing the dbt MCP URL" />

> 3. The tool will redirect you to dbt to complete the OAuth consent flow.

Author: ```suggestion
3. The tool will redirect you to dbt to complete the OAuth consent flow where you can approve or deny individual [scopes](#scopes-and-consent].
```

    <Lightbox src="/img/docs/dbt-cloud/oauth-consent-screen.png" width="60%" title="OAuth consent screen showing requested scopes and project access" />

4. The tool will be added to the **Custom connectors** table and you can then connect it to dbt by clicking **Connect**.
   <Lightbox src="/img/docs/dbt-cloud/oauth-connectors-page.png" title="Adding a custom dbt connector in an AI tool's connector settings" />

> 5. That's it! You can now use the tool to connect to dbt. Ask your tool a question like "What is the total revenue for the last 30 days?" go on from there!

Author: ```suggestion
5. That's it 🎉! You can now use the tool to connect to dbt. Ask your tool a question like "What is the total revenue for the last 30 days?" go on from there!
```

Dynamically registered apps appear in the **Dynamically registered** table, which shows:

| Column | Description |
|--------|-------------|
| Name | The app name |
| Redirect URL | The OAuth callback URL the app registered |
| Created on | When the client was first registered |
| Last used on | When the client last made an authenticated request |

HOW TO REMOVE A DYNAMICALLY REGISTERED CLIENT??

### Manual registration

For apps that don't support dynamic registration (for example, Salesforce agent or custom-built integrations), an admin can manually register an OAuth client.

To add a manually registered integration:

1. Go to **Account settings** → **Integrations** → **App integrations**.
2. Click **+ Add integration**.
3. Complete the registration form. Each manually registered client receives a unique client ID for use with PKCE flows. No client secret is issued.

Manually registered clients appear in the **Manually registered** section.

> ## Use with remote MCP

Reviewer 2: we'll surface the full constructed MCP URL in the app integrations section, so we can tell users that they can find their MCP url here to set up their AI tool of choice
